gateway-sp-comms/src/management_switch/location_map.rs

```diff
@@ -614,7 +594,7 @@ mod tests {
             }
         }
 
-        async fn resolve(self, log: &Logger) -> Result<String, StartupError> {
+        async fn resolve(self, log: &Logger) -> Result<String, String> {
```

Comment:
I'm a bit meh on string errors, but it's probably fine.

Reply:
That's fair, I am too. Does it help that `String` as the error type is private only, and these all get mapped to [`Error::DiscoveryFailed { reason: String }`](https://github.com/oxidecomputer/omicron/blob/e8d9458ec5cfdc7f61a91985d5cda42a7df1870f/gateway-sp-comms/src/management_switch.rs#L200) before leaving the crate? Or would you prefer that `reason` to be a nested error type itself?

Reply:
Ah, that's good. It's fine. We've got bigger fish to fry :)